File: java/Card.java
public class Card {
    private final String face;
    private final String suit;
    private double handValue; // For sorting within the hand

    public Card(String face, String suit) {
        this.face = face;
        this.suit = suit;
        this.handValue=0.0;
    }

    public String getFace() {
        return face;
    }

    public String getSuit() {
        return suit;
    }

    public double faceValue() {
        switch (face) {
            case "2": return 2;
            case "3": return 3;
            case "4": return 4;
            case "5": return 5;
            case "6": return 6;
            case "7": return 7;
            case "8": return 8;
            case "9": return 9;
            case "10": return 10;
            case "J": return 11;
            case "Q": return 12;
            case "K": return 13;
            case "A": return 14;
            default: return 0;
        }
    }
    
    public double suitValue(){
        switch (suit){
            case "D": return .1;
            case "C": return .2;
            case "H": return .3;
            case "S": return .4;
            default: return 0;
        }
    }

    public void setHandValue(double handValue) {
        this.handValue += handValue;
    }

    public double getHandValue() {
        return handValue;
    }

    @Override
    public String toString() {
        return face + suit;
    }
}

File: python/Card.py
class Card:
    def __init__(self, face, suit):
        self.face = face
        self.suit = suit
        self.hand_value = 0.0  # For sorting within the hand

    def get_face(self):
        return self.face

    def get_suit(self):
        return self.suit

    def face_value(self):
        face_values = {
            "2": 2, "3": 3, "4": 4, "5": 5,
            "6": 6, "7": 7, "8": 8, "9": 9,
            "10": 10, "J": 11, "Q": 12, "K": 13, "A": 14
        }
        return face_values.get(self.face, 0)

    def suit_value(self):
        suit_values = {
            "D": 0.1, "C": 0.2, "H": 0.3, "S": 0.4
        }
        return suit_values.get(self.suit, 0)

    def set_hand_value(self, hand_value):
        self.hand_value += hand_value

    def get_hand_value(self):
        return self.hand_value

    def __str__(self):
        return f"{self.face}{self.suit}"